Add prepare-pipelines guidance for first on-board services in mgmt-review workflow (#26678)
Azure/azure-sdk-for-go: .github/workflows/mgmt-review.md

### Step 1 — Gather information

1. Fetch PR details and changed files using GitHub MCP tools.
2. Identify the module path from the changed files (e.g., `sdk/resourcemanager/<service>/arm<package>/`).
3. Determine if this is a **first on-board service** (first beta version): check whether the PR adds a new `ci.yml` file under the module path (i.e., `ci.yml` appears in the changed files with status `added`). If so, record this for the Step 5 comment.

### Step 2 — Wait for all pipeline checks to complete

Before analyzing CI results, ensure all pipeline checks have finished running.

1. Fetch **check runs** for the PR head commit. Find the `go - pullrequest` parent check and its child jobs (`go - pullrequest (Build <job_name>)`). These are **Azure DevOps pipeline** results — do NOT call `get_job_logs` (returns 404).

> 2. If the `go - pullrequest` parent check or any of its child jobs have a `status` of `queued` or `in_progress`, poll check runs periodically (e.g., every 60 seconds) until **all** pipeline checks reach a **completed** status. If checks have not completed within **30 minutes**, stop waiting and analyze whatever results are available — note incomplete checks in the Step 5 comment.

Reviewer: We shall not design GitHub agentic workflows as long‑running operations.  Can we trigger this pipeline when all checks already completed.

Reviewer: Some best practise suggested by copilot:

Avoid:

single workflow doing everything end-to-end
workflows that wait or “loop” internally

Reviewer 2: Do you mean we should manually ensure the pipeline finish when we trigger this workflow?

Reviewer: I think so.  If some pipeline/check is still running, no need to wait till it finish.  People could easily trigger the workflow again.

Reviewer 3: Agree. Similarly, in the past when we have created solutions that poll check_runs instead of check_suites they tend to use up our github api quota and result in throttling for other actions/calls across the repo.

Reviewer 2: @copilot Change this LRO to just checking if the pipeline finish. If not, remind the user to re-trigger the check after the pipeline finish.

Reviewer 2: @lirenhe  @benbp Refined the instruction. Could you review again?

### Step 1 — Gather information

1. Fetch PR details and changed files using GitHub MCP tools.
2. Identify the module path from the changed files (e.g., `sdk/resourcemanager/<service>/arm<package>/`).
3. Determine if this is a **first on-board service** (first beta version): check whether the PR adds a new `ci.yml` file under the module path (i.e., `ci.yml` appears in the changed files with status `added`). If so, mark this PR as a first on-board service for Step 2.

### Step 2 — Note first on-board service

> **This step only applies when Step 1 identified a first on-board service.** Skip this step entirely otherwise.

> For a brand-new service, CI pipelines do not exist yet. The service team must create them by commenting `/azp run prepare-pipelines` on the PR. Record that this is a first on-board service so that the Step 6 comment includes instructions for the service team. This does **not** block the remaining steps — proceed to Step 3 immediately.

Reviewer 2: ```suggestion
For a brand-new service, release pipelines do not exist yet. The service team must create them by commenting `/azp run prepare-pipelines` on the PR. Record that this is a first on-board service so that the Step 6 comment includes instructions for the service team. This does **not** block the remaining steps — proceed to Step 3 immediately.
```

### Step 1 — Gather information

1. Fetch PR details and changed files using GitHub MCP tools.
2. Identify the module path from the changed files (e.g., `sdk/resourcemanager/<service>/arm<package>/`).
3. Determine if this is a **first on-board service** (first beta version): check whether the PR adds a new `ci.yml` file under the module path (i.e., `ci.yml` appears in the changed files with status `added`). If so, mark this PR as a first on-board service for Step 2.

### Step 2 — Prepare pipelines for first on-board service

> **This step only applies when Step 1 identified a first on-board service.** Skip this step entirely otherwise.

For a brand-new service, CI pipelines do not exist yet. They must be created before pipeline results can be checked.

> 1. Comment `/azp run prepare-pipelines` on the PR using `add_comment` to trigger pipeline creation.

Reviewer: Can we put this into comment as a required step for service team to handle?  Besides code review and adding some comments, I think the agentic work should not do any additional effort.

Reviewer 2: I prefer to auto-do it. This is somehow a shortage for our ci.

Reviewer 2: @copilot Change to leave instruction in step 6 instead of do it. And this does not block the other checks.

> 2. Wait for the `prepare-pipelines` check run to appear on the PR head commit. Poll check runs periodically (e.g., every 30 seconds) until a check whose name contains `prepare-pipelines` appears and reaches a **completed** status. If the check has not appeared or completed within **10 minutes**, stop waiting and record a timeout — it will be reported in the Step 6 comment.

Reviewer 2: Change timeout to 20mins

3. If `prepare-pipelines` fails, record the failure — it will be reported in the Step 6 comment.
4. After `prepare-pipelines` completes successfully, the CI validation pipeline (`go - pullrequest`) will be triggered automatically. Proceed to Step 3.

### Step 3 — Wait for all pipeline checks to complete

Before analyzing CI results, ensure all pipeline checks have finished running.

1. Fetch **check runs** for the PR head commit. Find the `go - pullrequest` parent check and its child jobs (`go - pullrequest (Build <job_name>)`). These are **Azure DevOps pipeline** results — do NOT call `get_job_logs` (returns 404).

> 2. If the `go - pullrequest` parent check or any of its child jobs have a `status` of `queued` or `in_progress`, poll check runs periodically (e.g., every 60 seconds) until **all** pipeline checks reach a **completed** status. If checks have not completed within **20 minutes**, stop waiting and analyze whatever results are available — note incomplete checks in the Step 6 comment.

Reviewer 2: Change timeout to 30mins

For failures not covered above, reference the [troubleshooting guide](https://github.com/Azure/azure-sdk-for-go/blob/main/documentation/development/TROUBLESHOOTING.md).

### Step 5 — Post a comment

Post **exactly one** PR comment via `add_comment`. Include the marker `<!-- gh-aw-workflow-id: mgmt-review -->` in the body.

**If pipeline checks have not completed** (detected in Step 2) → post only:

```markdown
## ⏳ Pipeline Checks Still Running

The `go - pullrequest` pipeline checks have not completed yet. Analysis cannot proceed until all checks finish.

**Action required:** Re-trigger this workflow by removing and re-adding the `mgmt-review-needed` label after the pipeline checks have completed.
```

**If all checks completed, nothing blocks, and this is NOT a first on-board service** → post only:

```
## PR is ready to merge
```

**If nothing blocks but this IS a first on-board service** (detected in Step 1) → post:

```markdown
## PR is ready to merge

## ⚠️ First On-Board Service — Pipeline Setup Required

This PR adds a new service. Release pipelines have not been created yet.

> **Action required:** Comment `/azp run prepare-pipelines` on this PR to create the release pipelines.

Reviewer 3: This does get run daily and will pick up new release pipelines after merge. So this step is no longer required, unless a user is expecting to be able to use their release pipelines immediately.

## PR is ready to merge
```

**If this is a first on-board service** (detected in Step 1), include this block in the comment:

> 

Reviewer 2: @copilot apply changes based on [this feedback](https://github.com/Azure/azure-sdk-for-go/pull/26678#discussion_r3153213927)

## PR is ready to merge
```

**If this is a first on-board service** (detected in Step 2), include this block in the comment:

```markdown
## ⚠️ First On-Board Service — Pipeline Setup Required

> This PR adds a new service. CI pipelines have not been created yet, so CI checks may be missing or incomplete.

Reviewer 2: ```suggestion
This PR adds a new service. Release pipelines have not been created yet.
```

> **Action required:** Comment `/azp run prepare-pipelines` on this PR to create the CI pipelines. After the `prepare-pipelines` check completes successfully, the CI validation pipeline (`go - pullrequest`) will be triggered automatically.

Reviewer 2: ```suggestion
**Action required:** Comment `/azp run prepare-pipelines` on this PR to create the release pipelines.
```
